src/main/scala/mesosphere/marathon/api/v2/json/V2Group.scala

```diff
@@ -36,4 +42,76 @@ object V2Group {
       groups = group.groups.map(V2Group(_)),
       dependencies = group.dependencies,
       version = group.version)
+
+  implicit val v2GroupValidator: Validator[V2Group] = validator[V2Group] { group =>
+    group.id is valid
+    group.apps is valid
+    group.groups is valid
+    group.groups is noAppsAndGroupsWithSameName(group.apps)
```

Comment:
The recursion is in group.groups or?
I still think you do not have to pass apps and make the validator a val. Obviously I miss something. Let's discuss f2f